### ISSUE TITLE
Sprint Poker: GitLab v1: Issues functionality 

### ISSUE THREAD

**Issue by nickoferrall** [ParabolInc/parabol#6088]
Predecessor: #6034 

### AC:

- [ ] As the user scrolls the list, more results are loaded
- [ ] Clicking an item adds the issue to the scope
- [ ] Each issue shows the issue number in the subtitle
- [ ] The select all functionality works
- [ ] Issues are sorted deterministically. We could use the `lastActivityAt` field

**Comment by mattkrick** [comment]
From 1:1

- As a first pass, we only allow 2 search criteria: search (ie text in title & description) and project (which is an array of projectIds selectable from the dropdown)
- the problem is the default query-- gitlab doesn't allow us to do a search among issues without specifying a project
- since GitLab provides a GraphQL endpoint, we'd like to use that without writing any custom code on our backend

```gql
query getIssuesByProjectIdAndSearch($search: String, $projectIds: [ID!], $isProjectFiltered: Boolean!, $first: Int, $after: String) {
  # If they have selected at least 1 project filter, then use that filter
  projects(ids: $projectIds) @include(if: $isProjectFiltered) {
    ...nodeSearch_nodes
  }
  # If they haven't selected a project to filter on, filter on the 3 most recently updated projects
  defaultProjects: projects(membership: true, first: 3, sort: "UPDATED_DESC") @skip(if: $isProjectFiltered) {
    ...nodeSearch_nodes
  }
}

fragment nodeSearch_nodes on ProjectConnection {
  nodes {
    issues(
      includeSubepics: true
      state: opened
      search: $search
      sort: UPDATED_DESC
      first: $first
      after: $after
    ) {
      nodes {
        id
        descriptionHtml
        title
      }
    }
  }
}

```

Out of scope:
- Currently the only way to see the project filters is to open the filter dropdown. this probably isn't great. we should see active filters without having to click a dropdown box. Nick says, "maybe we put little chits below the search input field?" CC @enriquesanchez
![Screen Shot 2022-02-23 at 10 51 35 AM](https://user-images.githubusercontent.com/5514175/155387432-18acbe8f-c97c-484e-a490-3756b007c8e8.png)

**Comment by enriquesanchez** [comment]
> Nick says, "maybe we put little chits below the search input field?" 

This is a great idea.

One thing that bothers me here is the lack of a border for the search field. It makes it difficult to correlate the field with the filter button. We could add a simple border around the whole thing and increase the clarity of the component, something like this:

<img width="854" alt="Captura de Pantalla 2022-02-24 a la(s) 5 36 41 p m" src="https://user-images.githubusercontent.com/3276087/155625867-f3a9f369-b615-458f-b984-ee4c6161ba47.png">

The filter button feels more integrated to the search field, and if there are any filters active, we can highlight it in blue. Also, now that we have more visual structure we can simply add a text string below the field indicating the active filters.

**Comment by nickoferrall** [comment]
@enriquesanchez I like it! The border looks great, and hopefully, the blue filter button draws more attention to that functionality. Text instead of chits sounds good too.

Two questions and then I'll create an issue for this:

1. If there are many filters, do you think the text should wrap to two or multiple lines or we add an ellipsis if it gets to the end of the first row?
2. In GitHub, we include the filters in the search query whereas in Jira and GitLab, we don't. I'm thinking we should add this design to all integrations, including Parabol tasks, but only show the "Current filters" text in Jira & GitLab. What do you think?

**Comment by enriquesanchez** [comment]
Oh, this are great questions 🙌

> If there are many filters, do you think the text should wrap to two or multiple lines or we add an ellipsis if it gets to the end of the first row?

I think we can display something like the first 3, and then finish with: `...and 9 more.` What do you think? I feel this is enough signal for users to know that the results are being filtered, and if they need more detail they can use the filter dropdown for more?

> but only show the "Current filters" text in Jira & GitLab

Can we not list the filters being applied? Is this a limitation of how the integration is set up¿

**Comment by nickoferrall** [comment]
> I think we can display something like the first 3, and then finish with: ...and 9 more. 

Sounds good to me!

> Can we not list the filters being applied?

We can, but the GitHub integration works by sending a search string that contains the repos we're filtering by. I think we've got a few options:
1. Keep the current design where the query string includes the repos, and don't include the "Current filters" below the search input
2. Show the repos in the query string _and_ show the "Current filters"
3. Show the "Current filters" and remove the repos from the query string. The problem with this is we want Parabol's GitHub integration to feel just like GitHub. If users can write `is:issue` it may feel unintuitive that they can't then write `repo:parabolinc/parabol`.

My preference is the first option. Parabol's GitHub integration will continue to feel just like GitHub, and we won't be repeating any information (option 2). The downside is it'll be inconsistent with the Jira & GitLab integrations which will show the "Current filters".

![Screenshot 2022-02-28 at 17 37 05](https://user-images.githubusercontent.com/39854876/156031166-1628b94d-3d4c-472b-a5e7-3b25e961a0a0.png)

**Comment by enriquesanchez** [comment]
Got it, thanks for the explanation, @nickoferrall 

> My preference is the first option. Parabol's GitHub integration will continue to feel just like GitHub, and we won't be repeating any information (option 2).

Agreed, option one sounds reasonable to me 👍 

> The downside is it'll be inconsistent with the Jira & GitLab integrations which will show the "Current filters".

I don't think this will be an issue considering that most users either have one integration or the other. So any possible confusion due to inconsistency should be minor, if not none at all.

**Comment by nickoferrall** [comment]
Makes sense! I've created an issue for this and added it to the GitLab goal: https://github.com/ParabolInc/parabol/issues/6138
